libraries/abstractions/common_io/test/test_iot_i2c.c

```diff
@@ -70,16 +68,13 @@ typedef struct CallbackParam
  * framework invoking these tests */
 /*-----------------------------------------------------------*/
 
-uint8_t uctestIotI2CSlaveAddr = 0;           /**< The slave address to be set for the I2C port. */
-uint8_t uctestIotI2CInvalidSlaveAddr = 0xFF; /**< The slave address to be set for the I2C port. */
-uint8_t uctestIotI2CDeviceRegister = 0;      /**< The device register to be set for the I2C port. */
-uint8_t uctestIotI2CWriteVal = 0;            /**< The write value to write to device. */
-uint8_t uctestIotI2CInstanceIdx = 0;         /**< The current I2C test instance index */
-uint8_t uctestIotI2CInstanceNum = 1;         /**< The total I2C test instance number */
-
-uint8_t ucAssistedTestIotI2CSlaveAddr = 0;   /**< The slave address to be set for the assisted test. */
-
-extern IotI2CHandle_t gIotI2cHandle[ testIotI2C_HANDLE_NUM ];
+uint16_t uctestIotI2CSlaveAddr = 0;             /**< The slave address to be set for the I2C port. */
```

Comment:
Why have these slave address variables been changed to `uint16_t`?

Reply:
The API defines slave address as an uint16_t and the ioctl calls expect the slave address to be passed as a uint16_t. This would be required for the test to support 10-bit addresses if ever to test for that (the API is OK with that if the HW is). 

As the ioctl expected a uint16_t, sending in a uint8_t resulted in an unknown upper half which  could throw unexpected failures depending on how the underlying port checks the range. 

For example, a port checking that the address is <= 0x7F because it only supports 7-bit addresses could be thrown of by someone sending a uint8_t with a valid value. This as the deference of the pointer, expecting it to be a uint16_t, would include the byte next to it as well.

Reply:
I agree with this change. uint16_t is right.

Reply:
This makes sense, but the comment on line 235 explicitly says that addresses are 7 bits, so this is a little confusing.  I'd consider deleting the "7 bit" part of that comment.